src/Tribe/Main.php

```diff
@@ -1398,6 +1400,121 @@ public function fix_all_day_events( $old_value, $new_value ) {
 			$wpdb->query( $fix_end_dates );
 		}
 
+		/**
+		 * Selects events to be moved to trash or permanently deleted.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $field_name - The name of the field that will be updated (trashPastEvents or deletePastEvents)
+		 * @param int    $old_value  - The default/existing number of months: this is the existing value used
+		 *                           to purge all events that ended before $old_value months
+		 * @param int    $new_value  - The updated number of months: this is the value chosen by user to purge all
+		 *                           events that ended before $new_value months
+		 *
+		 * @return array $post_ids   - an array of event Post_IDs with the Event End Date older than $new_value
+		 */
+		public function select_events_to_delete( $field_name, $old_value, $new_value ) {
+			$default_value = null;
+
+			// Prevent notices for undefined indexes
+			if ( empty( $old_value[ $field_name ] ) ) {
+				$old_value[ $field_name ] = $default_value;
+			}
+
+			if ( empty( $new_value[ $field_name ] ) ) {
+				$new_value[ $field_name ] = $default_value;
+			}
+			// If the field value is changed to 'Disabled', we can exit the execution here
+			if ( $new_value[ $field_name ] == $default_value ) {
+				return;
+			}
+			// If the value is not changed, we can exit the execution here
+			if ( $old_value[ $field_name ] == $new_value[ $field_name ] ) {
+				return;
+			}
+
+			global $wpdb;
+
+			$event_month_cutoff = $new_value[ $field_name ];
+
+			$sql = "SELECT post_id
+					FROM {$wpdb->posts} AS t1
+					INNER JOIN {$wpdb->postmeta} AS t2 ON t1.ID = t2.post_id
+					WHERE t1.post_type = %s
+						AND t2.meta_key = '_EventEndDate'
+						AND t2.meta_value <= DATE_SUB( CURDATE(), INTERVAL %s MONTH )";
+
+			$args = [
+				'post_type'  => self::POSTTYPE,
+				'meta_value' => $event_month_cutoff,
+			];
+
+			/**
+			 * This pair of filters allows users to manipulate the cleanup query
+			 *
+			 * @param string $sql  - the Query statement
+			 * @param array  $args - The array of variables to substitute into the query's placeholders
+			 *
+			 * @since TBD
+			 */
+			$sql  = apply_filters( 'tribe_events_delete_old_events_sql', $sql );
+			$args = apply_filters( 'tribe_events_delete_old_events_sql_args', $args );
+
+			// Returns an array of Post IDs (events) that ended before a specific date
+			$post_ids = $wpdb->get_col( $wpdb->prepare( $sql, $args ) );
+
+			$new_value[ $field_name ] = $default_value;
```

Comment:
Wondering if we could delete this line? Since the method returns on the next statement (and it returns a different var), not sure this is needed.

Reply:
Oops! I was testing other alternatives locally and forgot to delete this line. I'm removing it now, thanks Barry!